src/duckdb.f90

```diff
@@ -917,26 +918,95 @@ end subroutine duckdb_data_chunk_set_size
     ! =========================================================================
 
     ! DUCKDB_API duckdb_logical_type duckdb_vector_get_column_type(duckdb_vector vector);
+    function duckdb_vector_get_column_type(vector) bind(c, name='duckdb_vector_get_column_type') result(res)
+      import :: duckdb_vector, duckdb_logical_type
+      type(duckdb_vector) :: vector
+      type(duckdb_logical_type) :: res
+    end function duckdb_vector_get_column_type
 
     ! DUCKDB_API void *duckdb_vector_get_data(duckdb_vector vector);
+    function duckdb_vector_get_data(vector) bind(c, name='duckdb_vector_get_data') result(res)
+      import :: duckdb_vector, duckdb_result
+      type(duckdb_vector) :: vector
+      type(duckdb_result) :: res
+    end function duckdb_vector_get_data
 
     ! DUCKDB_API uint64_t *duckdb_vector_get_validity(duckdb_vector vector);
```

Comment:
not sure what to do with this function that returns a uint64_t array. by now have left it as a c_ptr.